src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -200,6 +200,18 @@ ss::future<response_ptr> alter_partition_reassignments_handler::handle(
     log_request(ctx.header(), request);
     alter_partition_reassignments_response resp;
 
+    if (!config::shard_local_cfg().kafka_enable_partition_reassignment()) {
+        vlog(
+          klog.info,
+          "Rejected alter partition reassignment request: API is disabled. See "
+          "`kafka_enable_partition_reassignment` configuration option");
+        resp.data.error_code = error_code::invalid_replica_assignment;
```

Comment:
Would `invalid_config` or `invalid_request` make more sense as an error code here?

Reply:
I tried invalid_request, but the error message that was printed by the client was weird.

Reply:
Makes sense, I'm fine with invalid_replica_assignment then. If a user has a problem with it, then we can change.